src/sp/protocol/reqrep0/req.c

```diff
@@ -190,18 +192,42 @@ req0_pipe_init(void *arg, nni_pipe *pipe, void *s)
 	return (0);
 }
 
+static void req0_pipe_insert_by_prio(nni_list *ready_pipes, req0_pipe *p){
+	req0_pipe *p2 = NULL;
+	int sndprio = p->sndprio;
+
+	if(sndprio < 1){
+		return nni_list_append(ready_pipes, p);
+	}
+
+    NNI_LIST_FOREACH(ready_pipes, p2) {
+        if (sndprio > p2->sndprio) {
+            nni_list_insert_before(ready_pipes, p, p2);
+            return ;
+        }
+    }
+    if (p2 == NULL) {
+        nni_list_append(ready_pipes, p);
+    }
+}
+
 static int
 req0_pipe_start(void *arg)
 {
 	req0_pipe *p = arg;
 	req0_sock *s = p->req;
-
+	int prio = 0;
+	
 	if (nni_pipe_peer(p->pipe) != NNG_REQ0_PEER) {
 		return (NNG_EPROTO);
 	}
-
```

Comment:
This option should be a generic option, not just on dialers, but on end points, and then configured on the pipe by the end point.

I also think that we should expand this to RECV priority as well, so that we can use the same option for REP to give preference to some peers.  I would like to think on this a bit.

Reply:
Hi, in the nanomsg code, I find the ep option when I call the nn_connect function, and then copy the socket option(ep_template) to pipe .
But in the nng code, I can't find the ep .  Are the dialer and listener eps? If so, I will add option on the listener also or only add option on the socket, and then copy this option to dialer or listener on creating them.

Reply:
I have added SNDPRIO to listener also.